docs/source/customizations/overview.mdx

```diff
@@ -23,98 +23,182 @@ The Apollo Router supports the following customization types:
-
-executionService --"<b>22.ExecutionResponse</b>"--> executionPlugins;
-executionPlugins --"<b>23.ExecutionResponse</b>"--> supergraphService;
-
-supergraphService --"<b>24.SupergraphResponse</b>"--> supergraphPlugins;
-supergraphPlugins --"<b>25.SupergraphResponse</b>"--> routerService;
-
-routerService --"<b>26.RouterResponse</b>"--> routerPlugins;
-routerPlugins --"<b>27.RouterResponse</b>"--> httpServer;
-
-httpServer --"<b>28.</b> HTTP response"--> client;
-
-
-class client,subgraph1,subgraph2 secondary;
+    subgraph router["Apollo Router"]
+        direction LR
+        httpServer("HTTP server")
+        subgraph routerService["Router Service"]
+            routerPlugins[[Router plugins]];
+        end
+        subgraph " "
+            subgraph supergraphService["Supergraph Service"]
+                supergraphPlugins[[Supergraph plugins]];
+            end
+            queryPlanner("Query Planner");
+        end
+        
+        
+        subgraph executionService["Execution Service"]
+            executionPlugins[[Execution plugins]];
+        end
+
+        subgraph subgraphService["Subgraph Services"]
+            subgraph service1["Subgraph Service A"]
+            subgraphPlugins1[[Subgraph plugins]];
+            end
+            subgraph service2["Subgraph Service B"]
+            subgraphPlugins2[[Subgraph plugins]];
+            end            
+        end
+    end;
+subgraphA[Subgraph A];
+subgraphB[Subgraph B];
+
+client --"1. HTTP request"--> httpServer;
+httpServer --"2. <code>RouterRequest</code>"--> routerService;
+routerService --"3. <code>SupergraphRequest</code>"--> supergraphService
+supergraphService --"4. Query"--> queryPlanner;
+queryPlanner --"5. Query plan"--> supergraphService;
+supergraphService --"6. <code>ExecutionRequest</code>"--> executionService;
+
+executionService --"7a. <code>SubgraphRequest</code>"--> service1;
+executionService --"7b. <code>SubgraphRequest</code>"--> service2;
+
+service1 --"8a. HTTP request"--> subgraphA;
+service2 --"8b. HTTP request"--> subgraphB;
 ```
 
----
+1. The router receives a client request at an HTTP server. 
+2. The HTTP server transforms the HTTP request into a `RouterRequest` containing HTTP headers and the request body as a stream of byte arrays.
+3. The router service receives the `RouterRequest`. It handles Automatic Persisted Queries (APQ), parses the GraphQL request from JSON and calls the supergraph service with the resulting `SupergraphRequest`.
+4. The supergraph service calls the query planner with the GraphQL query from the `SupergraphRequest`.
+5. The query planner returns a query plan for most efficiently executing the query.
+6. The supergraph service calls the execution service with an `ExecutionRequest`, made up of `SupergraphRequest` and the query plan.
+7. For each fetch node of the query plan, the execution service creates a `SubgraphRequest` and then calls the respective subgraph plugins and service.
```

Comment:
the issue in separating the request and response sides here is that in the execution service, subgraph requests can happen both in parallel and in sequence, because one subgraph response's data can be used in another subgraph request